src/impression.js

```diff
@@ -102,14 +104,24 @@ function trackImpressionFromClickParam(win, viewer, resolveImpression) {
   }
 
   viewer.whenFirstVisible().then(() => {
+    viewer.sendMessageAwaitResponse('getReplaceUrl', undefined).then(
```

Comment:
Should we limit to trusted viewers?

Reply:
I limited it trusted viewers or 'alp' experiment. Since viewer will also need to provide `replaceUrl` `caps` I think it's good enough.